core/src/main/java/org/apache/carbondata/core/locks/AlluxioFileLock.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.carbondata.core.locks;
+
+import java.io.DataOutputStream;
+import java.io.IOException;
+
+import org.apache.carbondata.common.logging.LogServiceFactory;
+import org.apache.carbondata.core.datastore.filesystem.AlluxioCarbonFile;
+import org.apache.carbondata.core.datastore.impl.FileFactory;
+import org.apache.carbondata.core.metadata.AbsoluteTableIdentifier;
+import org.apache.carbondata.core.util.path.CarbonTablePath;
+
+import org.apache.log4j.Logger;
+
+/**
+ * This class is used to handle the S3 File locking.
+ * This is acheived using the concept of acquiring the data out stream using Append option.
+ */
+public class AlluxioFileLock extends AbstractCarbonLock {
```

Comment:
why not use `HdfsFileLock` directly?, this class looks exactly same as `HdfsFileLock`

Reply:
We cannot use HDFSFileLock..the reason is we cannot go back to HDFS, for file locking. I understand although the structure is copied from HDFS File..when we run our jobs on alluxio, we may not be using HDFS at all, because the default file system (fs.defaultFS) falls back to Alluxio..

Reply:
@chandrasaripaka What I meant was we can use the same HDFSFileLock class or extended class of HDFSFileLock as I don't see much difference in code of `AlluxioFileLock` and `HDFSFileLock`

Reply:
Understand , will just extend from that.